src/lib.rs

```diff
@@ -128,11 +126,9 @@ impl XorName {
     }
 
     /// Generate a random XorName
-    pub fn random() -> Self {
+    pub fn random<T: Rng>(rng: &mut T) -> Self {
```

Comment:
This is clearly a breaking change.

We could possibly keep random() unchanged and add another random_rng(), or possibly sample() method that takes Rng param.

I guess an argument can be made that it is correct/better to force callers to pass in the Rng.  So I'm not saying this is wrong, just inquiring about the rationale.

Reply:
A bit of history of the `random()` function to save others digging

https://github.com/maidsafe/xor_name/commit/20b9ad5aef133c4fdc019cc901123d8d79849bde - feat(random): Add random xorname func
https://github.com/maidsafe/xor_name/commit/0a8325226ac0cbc5b3e7b315faa64654f78d4d24 - fix(rng): Use OsRng for security

Both in in https://github.com/maidsafe/xor_name/pull/41